### PR TITLE
CAMS-376 Address Security Issues

### PR DESCRIPTION
**Pull request by mattstankey**
# Purpose

Addresses medium-risk findings from the Debtor Name Search Threat Model Analysis: missing backend input validation and unbounded input length. Implements defense-in-depth validation at both the
  controller and use-case layers.

# Major Changes

  common/src/cams/search-validators.ts — Field-level validators and a casesSearchPredicateSpec validation spec for CasesSearchPredicate, using the CAMS custom validation library. Includes:              
  - Validators for caseNumber, debtorName (2-200 chars, trimmed), limit (1-100), offset (>= 0), divisionCodes, chapters (7/11/12/13/15), caseIds, excludedCaseIds                                         
  - A form-level atLeastOneSearchCriterion validator                                                                                                                                                      
  - Centralized error messages

  backend/lib/controllers/cases/cases.controller.ts — Added controller-layer validation in searchCases() that validates request.body against casesSearchPredicateSpec before casting to
  CasesSearchPredicate. Throws BadRequestError with detailed messages on invalid input.

  backend/lib/use-cases/cases/case-management.ts — Added checkValidation() method and use-case-layer validation in searchCases() as a second security boundary.

# Testing/Validation

  common/src/cams/search-validators.test.ts — 72 unit tests covering all validators (valid/invalid inputs, edge cases, undefined handling) and full-spec integration tests.

  backend/lib/controllers/cases/cases.controller.test.ts — Added 10 new tests for search input validation (no body, no criteria, invalid case number, debtor name too short/long, invalid
  limit/offset/chapters, verification use case isn't called). Fixed 2 existing tests that had incorrect request setup.

  backend/lib/use-cases/cases/case-management.test.ts — Fixed one test that used an invalid case number format ('001' → '00-00001').

# Notes

Optional - capture notes for nuances or future work that could be done.

# Definition of Done:

- [ ] Code refactored for clarity: Developers can understand the work simply by reviewing the code
- [ ] Dependency rule followed: More important code doesn’t directly depend on less important code
- [ ] Development debt eliminated: UX and code aligns to the team’s latest understanding of the domain
- [ ] No regressions: Changes do not cause regression in related or unrelated areas of the application

## Summary by Sourcery

Add shared search validation for case search predicates and enforce it at both the controller and use-case layers to strengthen backend input validation.

New Features:
- Introduce reusable validation spec and field-level validators for CasesSearchPredicate in the shared CAMS validation library.

Bug Fixes:
- Correct case search tests that were using invalid case numbers or incorrect HTTP request setup.

Enhancements:
- Validate case search requests in the cases controller and use case, returning structured BadRequest errors when input is invalid.

Tests:
- Add comprehensive unit tests for the new search validators and controller search input validation, ensuring the use case is not invoked on invalid input.

### CODE REVIEW

File: backend/lib/controllers/cases/cases.controller.ts [hunk 1]
```diff
@@ -55,6 +58,13 @@ export class CasesController implements CamsController {
   }
 
   private async searchCases(request: CamsHttpRequest) {
+    const validationResult = validateObject(casesSearchPredicateSpec, request.body);
```
[the remark this answers is not in the document]
Comment (mattstankey) [reply backend/lib/controllers/cases/cases.controller.ts]: Updated, removed duplicate validation from the controller layer.
